src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -170,7 +163,8 @@
   [rules :- Rules
    request :- ring/Request
    name :- schema/Str]
-  (if-let [ { matched-rule :rule matches :matches } (some #(match? % request) rules)]
+  (if-let [{matched-rule :rule matches :matches}
+           (some #(match? % request) (sort-rules rules))]
```

Comment:
Looks like they do get sorted in `transform-code` anyway - but the pre-sorted value is what's stored in the TK service context rather than the value after sorting.  Maybe could just store the value after sorting and not have to do it again here?